cpp/include/cudf/io/parquet.hpp

```diff
@@ -535,6 +535,7 @@ class chunked_parquet_reader {
 
  private:
   std::unique_ptr<cudf::io::parquet::detail::chunked_reader> reader;
+  parquet_reader_options _options;
```

Comment:
A generic question for cuIO code: Can we store options inside reader/writer objects in other cuIO code too? (just curious, why never stored it inside reader objects).

Reply:
I think what @ttnghia did in https://github.com/rapidsai/cudf/blob/branch-24.06/cpp/src/io/orc/reader_impl.hpp#L171-L181 is a good option. I assume we haven't done this initially to avoid making a copy of the options objects.

Reply:
Re-orged the `reader::impl`similar to ORC to simplify the simplify the use of `parquet_reader_options`.